.github/workflows/test.yml

```diff
@@ -61,3 +58,11 @@ jobs:
       - run: make test-parallel
       - run: make uninstall clean test-parallel PARALLEL_CONN=1
       - run: make test-parallel PARALLEL_CONN=1
+
+      # Test upgrade last, since the new version's client will be preferred.
+      - if: ${{ matrix.upgrade_to != '' }}
+        name: Upgrade to ${{ matrix.upgrade_to }}
+        run: |
+          make install
+          apt-get install -qq -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" postgresql-${{ matrix.upgrade_to }} postgresql-server-dev-${{ matrix.upgrade_to }}
```

Comment:
Maybe worth a comment on what `--force-confdef` and `--force-confold` does/why they're needed? (I've no familiarity with it so no clue how obvious the reasons are)

Reply:
I'd from Peter E's original script. I searched around and it seems like they help prevent prompting, which we want, but I don 't really understand them, either.

Reply:
Added a comment in 4479938